posthog/temporal/workflows/squash_person_overrides.py

```diff
@@ -241,38 +216,50 @@ def iter_person_overides_to_delete(self) -> Iterable[SerializablePersonOverrideT
 
 
 @activity.defn
-async def prepare_dictionary(inputs: QueryInputs) -> str:
-    """Prepare the DICTIONARY to be used in the squash workflow.
+async def prepare_person_overrides(inputs: QueryInputs) -> None:
+    """Prepare the person_overrides table to be used in a squash.
 
-    We also lock in the latest merged_at to ensure we do not process overrides that arrive after
-    we have started the job.
-    """
-    activity.logger.info("Preparing DICTIONARY %s.%s", inputs.database, inputs.dictionary_name)
+    This activity executes two queries:
+    - First one is a DETACH TABLE to ensure no new data is ingested during the squash.
+    - Second one is a OPTIMIZE TABLE to ensure we assign the latest overrides for each old_person_id.
 
-    latest_created_at = sync_execute(SELECT_LATEST_CREATED_AT_QUERY.format(database=inputs.database))[0][0]
-    activity.logger.info("Populating underlying TABLE %s.%s_table", inputs.database, inputs.dictionary_name)
+    The activity to re-attach person_overrides should always be executed after this one.
+    """
+    activity.logger.info("Detaching %s.kafka_person_overrides", inputs.database)
     sync_execute(
-        CREATE_DICTIONARY_TABLE_QUERY.format(
-            database=inputs.database,
-            dictionary_name=inputs.dictionary_name,
-            user=inputs.user,
-            password=inputs.password,
-            cluster_name=inputs.cluster_name,
+        "DETACH TABLE {database}.kafka_person_overrides ON CLUSTER {cluster}".format(
```

Comment:
Thinking about it now I don't have too much of a reason for this belief but I generally prefer detaching the MVs rather than the Kafka tables.

I think my intuition is that the MV existing, it will still try to pull data from that Kafka table which will no longer "exist". The Kafka table in turn is just an initialized consumer that only starts running when the table is queried (i.e. a SELECT issued by the MV). So feels natural to me to leave the scaffolding and cut of the mechanism that pulls the data. I also like that with views I can use the safer `DETACH VIEW` which would prevent me from running `DETACH TABLE events` or something.

Although ultimately not much reason for this so certainly not a blocker.

Reply:
I can swap out to detaching the mat view. Makes sense to stop pulling data than detaching the consumer.